web/php/Actions/AccountProfileAction.php

```diff
@@ -226,28 +225,25 @@ public function saveAboutEvent($runData)
 
         $location = $pl->getParameterValue("location");
 
-        $profile->setRealName($realName);
-        $profile->setPronouns($pronouns);
+        $profile->real_name = $realName;
+        $profile->pronouns = $pronouns;
 
         // check date
         $d = getdate();
         if (checkdate((int) $birthdayMonth, (int) $birthdayDay, (int) $birthdayYear) && $birthdayYear < $d['year']) {
-            $profile->setBirthdayDay($birthdayDay);
-            $profile->setBirthdayMonth($birthdayMonth);
-            $profile->setBirthdayYear($birthdayYear);
+            // Pad with zeroes if needed.
+            if($birthdayMonth < 10) { $birthdayMonth = "0".$birthdayMonth; }
+            if($birthdayDay < 10) { $birthdayDay = "0".$birthdayDay; }
```

Comment:
You can use [`str_pad`](https://www.php.net/manual/en/function.str-pad.php) here instead:
```php
$birthdayMonth = str_pad($birthdayMonth, 2, '0', STR_PAD_LEFT);
$birthdayDay = str_pad($birthdayDay, 2, '0', STR_PAD_LEFT);
```
It works even if the input is an integer. Unfortunately the default is to pad right rather than left, which makes the call rather long.

Reply:
Nice, that sounds better.